js/modules/k6/http/response.go

```diff
@@ -137,6 +138,36 @@ func (res *HTTPResponse) Json() goja.Value {
 		default:
 			common.Throw(common.GetRuntime(res.ctx), errors.New("Invalid response type"))
 		}
+
+		if hasSelector {
+			result := gjson.GetBytes(body, selector[0])
+
+			if !result.Exists() {
+				return goja.Undefined()
+			}
+
+			switch result.Type {
```

Comment:
I looked at the gjson library and, I'm not sure, but I think we may be able to safely replace everything from the `switch` here to the end of the `if` (line 169 currently) with just `return common.GetRuntime(res.ctx).ToValue(result.Value())`. 

Goja's [`Runtime.ToValue()`](https://godoc.org/github.com/dop251/goja#Runtime.ToValue) method accepts `interface{}`, so goja would do type assertions anyway. And unless I'm mistaken, gjson's [`Result.Value()`](https://godoc.org/github.com/tidwall/gjson#Result.Value) method returns the fully-parsed hierarchical value, so there should be no need to use the stdlib's `json.Unmarshal()`?

Reply:
That's a good point, 
It really works and does not affect performance, at least on my laptop.
I'll do commit with this changes